### ISSUE TITLE
Parent window to host

### ISSUE THREAD

**Issue by mottosso** [pyblish/pyblish-qml#1]
Currently, the GUI is launched as a separate window, but we'd like it to appear as though it was a child of to the host.

This may require OS-specific commands.

On Windows, SetParent seems the obvious winner, though it seems to have some warts.
- https://mail.python.org/pipermail/python-win32/2011-May/011446.html
- http://srinikom.github.io/pyside-bz-archive/523.html

Something like this ought to do it for Maya on Windows.

``` python
from PySide import QtGui

widgets = dict((w.objectName(), w) for w in QtGui.QApplication.topLevelWidgets())
window = widgets['MayaWindow']
id = window.winId()

import ctypes
ctypes.pythonapi.PyCObject_AsVoidPtr.restype = ctypes.c_void_p
ctypes.pythonapi.PyCObject_AsVoidPtr.argtypes = [ctypes.py_object]

int_hwnd = ctypes.pythonapi.PyCObject_AsVoidPtr(id)
```

If a similar feature is available in Linux distributions (Wayland?) then that's where we'll go.

**Comment by mottosso** [comment]
Some thoughts on a potential solution to this.

I'm reluctant to try physically parenting the window to another process due to having read scare stories about it on the interwebs.

As an alternative, I'm thinking it may be more safe and equally user-friendly to simply introspect the external process, like it's current minimized state, size and position, and mirror this for our GUI.

The experience should be identical, but without the pitfalls of physical parenting.

## Implementation

We can get the HWND from a host via `QWidget.winId()`. With that, we can poll for changes, such as whether or not the window has been minimized, moved or resized.

http://docs.activestate.com/activepython/3.2/pywin32/win32gui.html

We'll use this information in the GUI code to "monkey do, monkey say". 

Another advantage of this approach is that we're safe from hosts crashing and vice versa; our app can't cause a host to crash. We can also choose to "detach" should be want our GUI to remain open during e.g. long running publishes.

**Comment by mottosso** [comment]
Implemented in #259
